scripts/find_busy_times.py

```diff
@@ -0,0 +1,130 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Computes the expected number of tests on TPUs at different times of day.
+
+Parses all the kubernetes test configs in a given directory. This script
+relies on the `activeDeadlineSeconds` field in the configs to tell how long
+a job is expected to last. If that field is overly generous in your test
+configs, then this script will overestimate the number of active jobs.
+
+This script outputs an HTML file to `job_frequency.html` with a table of
+days/times and the number of expected tests running at that time.
+
+The `chunk_minutes` flag determines granularity of the grid. For example if
+`chunk_minutes=10`, then the grid will show 10-minute intervals in the
+output grid.
+
+Example usage:
+
+pip3 install croniter
+python3 scripts/find_busy_times.py --dir_path=/Users/me/ml-testing-accelerators/k8s/us-central1/gen/* --chunk_minutes=15
+"""
+
+import datetime
+import glob
+import re
+import sys
+
+import croniter
+
+from absl import app
+from absl import flags
+
+FLAGS = flags.FLAGS
+flags.DEFINE_string('dir_path', None, 'Dir to search for test configs.')
+flags.DEFINE_integer('chunk_minutes', 10, 'Number of minutes per chunk.')
+
+_DEADLINE_REGEX = r'"?activeDeadlineSeconds"?: "?(\d+)"?'
```

Comment:
FYI, this might not work nicely if we set a deadline on individual pods in addition to the job deadline. You may be better off parsing the files with pyyaml and getting `spec.template.spec.activeDeadlineSeconds` directly. I'll leave that up to you.